client/components/marketing-survey/cancel-jetpack-form/index.tsx

```diff
@@ -0,0 +1,243 @@
+		setCancellationStep( steps.FEATURES_LOST_STEP );
+		setSurveyResponse( false );
+	};
+
+	// Record an event for Tracks
+	const recordEvent = ( name: string, properties = {} ) => {
+		const { purchase, flowType } = props;
+
+		recordTracksEvent( name, {
+			cancellation_flow: flowType,
+			product_slug: purchase.productSlug,
+			is_atomic: isAtomicSite,
+
+			...properties,
+		} );
+	};
+
+	// run on mount
+	useEffect( () => {
+		resetSurveyState();
+	}, [] );
+
+	// if isVisible changes
+	useEffect( () => {
+		if ( isVisible && cancellationStep === steps.INITIAL_STEP ) {
+			recordEvent( 'calypso_purchases_cancel_form_start' );
+		}
+	}, [ isVisible ] );
+
+	/**
+	 * Get possible steps for the survey
+	 */
+	const getAvailableSurveySteps = () => {
+		// this could vary based on the user data/ product
+		// for now, only showing two steps for all Jetpack plans and products
+		return [ steps.FEATURES_LOST_STEP, steps.CANCELLATION_REASON_STEP ];
+	};
+
+	const handleCloseDialog = () => {
+		props.onClose();
+		resetSurveyState();
+
+		// record tracks event
+		// sends the same event name as the main product cancellation form
+		recordEvent( 'calypso_purchases_cancel_form_close' );
+	};
+
+	const setSurveyStep = ( stepFunction: ( step: string, steps: string[] ) => string ) => {
+		const availableSteps = getAvailableSurveySteps();
+		const newStep = stepFunction( cancellationStep, availableSteps );
+
+		setCancellationStep( newStep );
+
+		// record tracks event
+		// since the steps used for the Jetpack cancellation flow are different, use a different event name for Tracks
+		recordEvent( 'calypso_purchases_cancel_jetpack_survey_step', { new_step: newStep } );
+	};
+
+	const clickNext = () => {
+		setSurveyStep( nextStep );
+	};
+
+	const clickPrevious = () => {
+		setSurveyStep( previousStep );
+	};
+
+	const onSubmit = () => {
+		// TODO: handle survey submission - probably using the existing submitSurvey method
+
+		// call back to the parent component to actually cancel the subscription
+		props.onClickFinalConfirm();
+
+		// record tracks event
+		// this uses the same event name as the main product cancellation form
+		// this way, all cancellation stats can be viewed together
+		recordEvent( 'calypso_purchases_cancel_form_submit' );
+	};
+
+	/**
+	 * Render the dialog buttons for the current step
+	 */
+	const renderStepButtons = () => {
+		const { disableButtons } = props;
+		const disabled = disableButtons;
+
+		// in most cases, we will have a previous and next
+		const close = {
+			action: 'close',
+			disabled: disabled,
+			label: translate( "I'll Keep It" ),
+		};
+		const next = {
+			action: 'next',
+			disabled: disabled,
+			label: translate( 'Continue Cancelling' ),
+			onClick: clickNext,
+		};
+		const prev = {
+			action: 'prev',
+			disabled,
+			label: translate( 'Previous Step' ),
+			onClick: clickPrevious,
+		};
+
+		const cancelText = translate( 'Cancel Now' );
+		const cancellingText = translate( 'Cancelling' );
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'Why cancelling?' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)